presto-main/src/main/java/com/facebook/presto/sql/planner/CanonicalPlanGenerator.java

```diff
@@ -830,6 +833,25 @@ public Optional<PlanNode> visitAggregation(AggregationNode node, Context context
         return Optional.of(canonicalPlan);
     }
 
+    @Override
+    public Optional<PlanNode> visitSequence(SequenceNode node, Context context)
+    {
+        node.getCteProducers().forEach(x -> x.accept(this, context));
```

Comment:
Do we need to visit producers?

Reply:
Yes, it's to trigger canonicalization of plan nodes under the cte producer, which will also need HBO to optimize the subquery.